finagle/buoyant/src/main/scala/com/twitter/finagle/buoyant/ExistentialStability.scala

```diff
@@ -56,27 +60,35 @@ object ExistentialStability {
         @volatile var current: VarUp[T] = null
         @volatile var exists = false
         unstable.states.respond {
-          case Activity.Ok(Some(t)) if current == null =>
+          case Activity.Ok(Some(t)) if current == null => synchronized {
```

Comment:
What are we synchronizing on here?  I think it's essentially `unstable.synchronized` since ExistentiaAct is a value class.  Really, we probably want to synchronize on the `Var.async` closure.  We could effectively do this by creating `val mu = new {}` alongside `current` and `exists` and then doing `mu.synchronized`

Reply:
Do you think it would better express intent to, say, bundle `current` and `exists` into an object and synchronize on _that_, instead?